homeassistant/components/ubus/device_tracker.py

```diff
@@ -116,29 +128,44 @@ def _update_info(self):
         if not self.success_init:
             return False
 
-        _LOGGER.info("Checking hostapd")
+        _LOGGER.info("Checking hostapd for host: %s", self.host)
 
-        if not self.hostapd:
+        if not self.wlans:
             hostapd = _req_json_rpc(self.url, self.session_id, "list", "hostapd.*", "")
-            self.hostapd.extend(hostapd.keys())
+            for name in hostapd.keys():
+                iwinfo = _req_json_rpc(
```

Comment:
Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices